components/centraldashboard/README.md

```diff
@@ -118,3 +122,34 @@ body > .Main-Toolbar {background-color: var(--primary-background-color)}
 .List > .item:not(:first-of-type) {border-top: 1px solid var(--border-color)}
 /* etc */
 ```
+
+## Client-Side Library
+Since the Central Dashboard wraps other Kubeflow components in an iframe, a
+standalone Javascript library is exposed to make it possible to communicate
+between the child pages and the dashboard.
+
+The library is available for import from `/dashboard_lib.bundle.js` within the
+cluster. When imported in this manner, a `centraldashboard` module is created in
+the global scope. To establish a channel for communication between the iframed
+page and the Dashboard, use the following example which would disable a
+`<select>` element and assign the value from the Dashboard's Namespace selector
+when it changes:
+
+```
+window.addEventListener('DOMContentLoaded', function (event) {
+    if (window.centraldashboard.CentralDashboardEventHandler) {
+        // Init method will invoke the callback with the event handler instance
+        // and a boolean indicating whether the page is iframed or not
+        window.centraldashboard.CentralDashboardEventHandler
+            .init(function (cdeh, isIframed) {
+                var namespaceSelector = document.getElementById('ns-select');
+                namespaceSelector.disabled = isIframed;
+                // Binds a callback that gets invoked anytime the Dashboard's
+                // namespace is changed
+                cdeh.onNamespaceSelected = function (namespace) {
+                    namespaceSelector.value = namespace;
+                };
+            });
+    }
+});
```

Comment:
A few questions:
- Why can't `window.centraldashboard` **be** the `CentralDashboardEventHandler` or simply `window.CentralDashboardEventHandler` (if this name is preferred)
  - Also since the scope for the SDK is to also have the child be able to pass messages back to central-dashboard, wouldn't it make more sense to have a different name, since EventHandling isn't the only task it does?
- The approach of `cdeh.onNamespaceSelected = fn` only allows users to link a single handler, instead of a linked list approach. Is this an intended use case for all sub-apps?
  - What I mean is that `cdeh.onNamespaceSelected(fn)` would allow users to bind multiple handlers
- Since we're calling the lib `SDK`, wouldn't it make sense to name all the objects accordingly?
  - `cedh` -> `sdk`
  - `CentralDashboardEventHandler` -> `CentralDashboard_SDK`
- Are we using `var` because of retroactive support?

These are just notes/questions, I am not trying to suggest we do things one way or another. I am just making sure it makes sense from a user perspective. With that here is what I was envisioning:
```js
window.addEventListener('DOMContentLoaded', e => {
    const CD_SDK = window.CentralDashboard_SDK
    if (!CD_SDK) return
    CD_SDK.init((sdk, isIframed) => {
        const namespaceSelector = document.getElementById('ns-select')
        namespaceSelector.disabled = isIframed
        // Binds a callback that gets invoked anytime the Dashboard's
        // namespace is changed
        sdk.onNamespaceSelected(namespace => {
            namespaceSelector.value = namespace
        })
    })
})
```

Reply:
> Why can't window.centraldashboard be the CentralDashboardEventHandler or simply window.CentralDashboardEventHandler (if this name is preferred)
Also since the scope for the SDK is to also have the child be able to pass messages back to central-dashboard, wouldn't it make more sense to have a different name, since EventHandling isn't the only task it does?

Webpack exports the library as a [UMD](https://github.com/umdjs/umd) and since there are multiple exports rather than a single default export, we have to access the class by its module-qualified name. For the time being, I didn't see a scope beyond sending and responding to events  in the near to intermediate term future. I'm also not great at naming so if you think there's a better name I'm open to it.

> The approach of cdeh.onNamespaceSelected = fn only allows users to link a single handler, instead of a linked list approach. Is this an intended use case for all sub-apps?
What I mean is that cdeh.onNamespaceSelected(fn) would allow users to bind multiple handlers

For simplicity's sake I just allowed for a single handler rather than multiple observers. I couldn't think of a scenario where the latter would be necessary especially since we're building this to fit a fairly narrow requirement at the moment.

> Since we're calling the lib SDK, wouldn't it make sense to name all the objects accordingly?
cedh -> sdk
CentralDashboardEventHandler -> CentralDashboard_SDK

If you feel strongly about including SDK in the name we can obviously make that change. To me, SDK implies a set of tools to facilitate development on a particular platform which I don't necessarily feel that we're offering. Rather, I think we're offering a lightweight coupling between disparate applications. In the future, perhaps we would need other types of library utilities that might be better served from their own class structure and this naming facilitates that.

> Are we using var because of retroactive support?
Yes, since the Jupyter application does not go through any type of transpilation, the example snippet in the README is plain old ES3 Javascript.

If you feel we're missing the mark and would like to pull back into a deeper design phase, I'm happy to oblige and perhaps follow a more directed approach for integrating with the namespace selector on the Jupyter notebook server property.